etl/steps/data/garden/war/2024-01-23/nuclear_weapons_treaties.py

```diff
@@ -0,0 +1,170 @@
+# Label for the status "Signatory", which denotes a country that endorses the treaty but does not legally commit to it.
+LABEL_AGREED = "Signed"
+# Label for the status "Ratification", "Accession" and "Succession", used for countries that legally commit to a treaty.
+# "Ratification" denotes a country that has ratified the treaty after signing it.
+# "Accession" denotes a country that has ratified the treaty without having signed it first.
+# "Succession" denotes a country that has inherited the status of a predecessor that has ratified the treaty.
+# NOTE: A priori it may be possible that a country inherits the status "Signatory" from a predecessor, but later on I
+# check that this is never the case.
+LABEL_COMMITTED = "Ratified"
+# Label for the exceptional status "Withdrawal", which denotes a country that has withdrawn from the legal commitment.
+LABEL_WITHDRAWN = "Withdrawn"
+# Label for all countries-years that are not posterior to either an agreement or a commitment.
+LABEL_NOT_CONSIDERED = "Not signed"
+
+# List of known withdrawals of any treaty.
+WITHDRAWALS = [{"treaty": "Comprehensive Nuclear-Test-Ban Treaty", "country": "Russia", "date": "2023-11-03"}]
+
+
+def run_sanity_checks(tb: Table) -> None:
+    for treaty in set(tb["treaty"]):
+        _tb = tb[tb["treaty"] == treaty].reset_index(drop=True)
+        # Check that countries that have the status "Succession" did not previously have only the status "Signatory".
+        # In other words, check that a country never inherits the "Signatory" status from a predecessor.
+        # If confirmed, we can safely treat "Succession" as the same status as "Accession" and "Ratification".
+        # NOTE: A country may have the status "Signatory", then change to a legally binding status, and then change to
+        # the status "Succession". This is the case of Serbia for the partial-test-ban treaty.
+        # But what I am checking is that a country cannot have only the status "Signatory" inherited from a predecessor.
+        countries_with_succession = set(_tb[_tb["status"] == "Succession"]["country"])
+        assert set(_tb[_tb["country"].isin(countries_with_succession)]["status"]) != {"Signatory"}
+        # By definition, a country with the status "Accession" is one that commits to the treaty without having a prior
+        # status "Signatory".
+        # Check that a country that has the status "Accession" does not have any prior status "Signatory".
+        countries_with_accession = set(_tb[_tb["status"] == "Accession"]["country"])
+        assert "Signatory" not in set(_tb[_tb["country"].isin(countries_with_accession)]["status"])
+    error = "The list of withdrawals has changed."
+    assert sorted(tb[tb["status"] == "Withdrawal"].drop(columns=["status"]).to_dict(orient="records")) == sorted(
+        WITHDRAWALS  # type: ignore
+    ), error
+
+
+def prioritize_status(statuses):
```

Comment:
Maybe you want to summarise this in some of the `description_processing` fields in metadata.

Reply:
Yes, I'll explain these assumptions in `description_processing`, in a separate PR.